client/app/bundles/course/material/folders/pages/FolderShow/index.tsx

```diff
@@ -65,6 +65,14 @@ const FolderShow: FC = () => {
 
   const headerToolbars: ReactElement[] = [];
 
+  if (folderId === undefined) {
+    window.history.replaceState(
```

Comment:
at the first glance, it's a bit confusing on why if `folderId` is `undefined`, then we redirect to `currFolderInfo.id` (it's not clear here how `currFolderInfo` is definitely our root folder, even though if we trace it further, we know that the controller `index` is called and the `currFolderInfo` is set to root folder).

To reduce the confusion (and potentially misunderstanding in the future), I suggest we make it clear that `currFolderInfo.id` is our root folder by doing this

```
const rootFolderId = currFolderInfo.id
window.history.replaceState({}, '', getWorkbinFolderURL(getCourseId(), currFolderInfo.id))
```